src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingRegistry.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.registry;
+
+import com.aws.greengrass.clientdevices.auth.exception.CloudServiceInteractionException;
+import com.aws.greengrass.clientdevices.auth.iot.Certificate;
+import com.aws.greengrass.clientdevices.auth.iot.IotAuthClient;
+import com.aws.greengrass.clientdevices.auth.iot.Thing;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+
+public class ThingRegistry {
+    // holds mapping of thingName to IoT Certificate ID;
+    // size-bound by default cache size, evicts oldest written entry if the max size is reached
+    static final Map<String, String> registry = Collections.synchronizedMap(
+            new LinkedHashMap<String, String>(RegistryConfig.REGISTRY_SIZE, 0.75f, false) {
+                @Override
+                protected boolean removeEldestEntry(Map.Entry eldest) {
+                    return size() > RegistryConfig.REGISTRY_SIZE;
+                }
+            });
+
+    private final IotAuthClient iotAuthClient;
+
+    @Inject
+    public ThingRegistry(IotAuthClient iotAuthClient) {
+        this.iotAuthClient = iotAuthClient;
+    }
+
+    /**
+     * Returns whether the Thing is associated to the given IoT Certificate.
+     * Returns locally registered result when IoT Core cannot be reached.
+     * TODO: add a separate refreshable caching layer for offline auth
+     *
+     * @param thing IoT Thing
+     * @param certificate IoT Certificate
+     * @return whether thing is attached to the certificate
+     */
+    public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate) {
+        try {
+            if (iotAuthClient.isThingAttachedToCertificate(thing, certificate)) {
+                registerCertificateForThing(thing, certificate);
+                return true;
+            } else {
+                clearRegistryForThing(thing);
+            }
+        } catch (CloudServiceInteractionException e) {
+            return isCertificateRegisteredForThing(thing, certificate);
+        }
+        return false;
+    }
+
+    /**
+     * Clears registry cache.
+     */
+    public void clear() {
```

Comment:
Unit tests use this method to clear the registry during tear-down.  Let me make it package-private.

Reply:
As a general rule, try to avoid adding methods for unit tests. Prefer creating new objects rather than "resetting" them. In this case I don't think it's necessary to clear anything since you're creating a new registry object each time, right?